File: 1/2020.10.01/860.Lemonade Change/python/solution.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        five = ten = 0
        for bill in bills:
            if bill == 5:
                five += 1
            elif bill == 10:
                if five == 0:
                    return False
                five -= 1
                ten += 1
            else:
                if ten > 0:
                    if five == 0:
                        return False
                    ten -= 1
                    five -= 1
                else:
                    if five < 3:
                        return False
                    five -= 3
        return True

File: 1/2020.10.01/860.Lemonade Change/java/solution.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        int five = 0, ten = 0;
        for (int bill : bills) {
            if (bill == 5) {
                five++;
            }else if (bill == 10) {
                if (five == 0) {
                    return false;
                }
                five--;
                ten++;
            } else if (ten > 0) {
                if (five == 0) {
                    return false;
                }
                ten--;
                five--;
            } else if (five < 3) {
                return false;
            } else {
                five -= 3;
            }
        }
        return true;
    }
}
